### PR TITLE
Cleanup nits

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Lots of gratuitous .gitignore files and some inconsistencies in the BPF test Cargo.tomls

#### Summary of Changes

- Remove the .giitignore files that are not needed
- Removed the CI check that enforces said gratuitous files
- Cleanup the BPF test programs Cargo.toml files

Fixes #

### CODE REVIEW

File: ci/nits.sh [hunk 1]
```diff
@@ -43,25 +43,4 @@ fi
...
-      gitignores_ok=false
-    fi
-    if ! grep -q -e '^/farf/$' "$dir"/.gitignore ; then
-      echo 'error: nits.sh "/farf/" apparently missing from '"$dir"'/.gitignore' >&2
-      gitignores_ok=false
-    fi
-  fi
-done
-"$gitignores_ok"
+fi
```
Comment (rob-solana) [inline ci/nits.sh]: no newline?

File: programs/bpf/rust/alloc/Cargo.toml [hunk 2]
```diff
@@ -19,5 +19,5 @@ program = ["solana-sdk/program"]
 default = ["program"]
 
 [lib]
-crate-type = ["cdylib"]
 name = "solana_bpf_rust_alloc"
+crate-type = ["cdylib"]
```
Comment (rob-solana) [inline programs/bpf/rust/alloc/Cargo.toml]: here, too?

**Comment by jackcmay** [comment]
Heh, I'm newline indifferent, @mvines seems on a mission to get rid of any wayward newlines.  @rob-solana Do either of you have a strong opinion either way or a reason that having or not having one will cause an issue?

**Comment by mvines** [comment]
I'm an always put a newline at the end of the file kind of guy myself.

**Comment by rob-solana** [comment]
my editor complains, as does github

there are a lot of poorly-written line-mode tools, libraries, interpreters, what-have-you that will ignore bytes at the end of a file as a non-line.

so I'm a fan of newlines at the end of files

**Comment by jackcmay** [comment]
Sounds good, I was confusing @mvines dislike of blank lines with new-lines

**Pull request merged**